core/src/main/java/org/infinispan/commands/read/AbstractDataCommand.java

```diff
@@ -7,30 +7,40 @@
 
 import org.infinispan.commands.DataCommand;
 import org.infinispan.commands.SegmentSpecificCommand;
+import org.infinispan.commands.remote.CacheRpcCommand;
 import org.infinispan.context.Flag;
 import org.infinispan.context.impl.FlagBitSets;
 import org.infinispan.marshall.protostream.impl.MarshallableObject;
 import org.infinispan.protostream.annotations.ProtoField;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.telemetry.InfinispanSpanAttributes;
+import org.infinispan.util.ByteString;
 
 /**
  * @author Mircea.Markus@jboss.com
  * @author Sanne Grinovero &lt;sanne@hibernate.org&gt; (C) 2011 Red Hat Inc.
  * @since 4.0
  */
-public abstract class AbstractDataCommand implements DataCommand, SegmentSpecificCommand {
+public abstract class AbstractDataCommand implements CacheRpcCommand, DataCommand, SegmentSpecificCommand {
```

Comment:
nitpicking, adding `CacheRpcCommand` is redundant. 
```suggestion
public abstract class AbstractDataCommand implements DataCommand, SegmentSpecificCommand {
```